### PR TITLE
#1303 fix mesage count issue

### LINKED ISSUE #1303: Message Count showing wrong
**Describe the bug**
Message read count not updating even if thread is already deleted and user don't have any messages in inbox.

**Support ticket links**
https://secure.helpscout.net/conversation/1157649807/72283/

### PR DESCRIPTION
**Pull request by chetansatasiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

1. We added a new option to repair the count in Backend here: BuddyBoss > Tools > Repair Community > Repair unread messages count on the site.

Fixes #1303  .

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Added a new option to repair the count in Backend here: BuddyBoss > Tools > Repair Community > Repair unread messages count on the site.

### CODE REVIEW

File: src/bp-messages/bp-messages-filters.php [hunk 1]
```diff
@@ -91,6 +91,7 @@
 add_action( 'groups_leave_group', 'bp_group_messages_remove_group_member_from_thread', 10, 2 );
 add_action( 'groups_remove_member', 'bp_group_messages_remove_group_member_from_thread', 10, 2 );
 
+add_filter( 'bp_repair_list', 'bp_messages_unread_count_repair_items' );
```
Comment (mehulkaklotar) [inline src/bp-messages/bp-messages-filters.php:94]:
```suggestion
add_filter( 'bp_repair_list', 'bp_messages_repair_items_unread_count' );
```

File: src/bp-messages/bp-messages-filters.php [hunk 2]
```diff
@@ -653,3 +654,59 @@ function bp_messages_remove_user_to_group_message_thread( $group_id, $user_id )
...
+
+/**
+ * Add message notification repair list item.
+ *
+ * @param $repair_list
+ *
+ * @since BuddyBoss 1.4.8
+ * @return array Repair list items.
+ */
+function bp_messages_unread_count_repair_items( $repair_list ) {
```
Comment (mehulkaklotar) [inline src/bp-messages/bp-messages-filters.php:666]:
```suggestion
function bp_messages_repair_items_unread_count( $repair_list ) {
```

File: src/bp-messages/bp-messages-filters.php [hunk 3]
```diff
@@ -653,3 +654,59 @@ function bp_messages_remove_user_to_group_message_thread( $group_id, $user_id )
...
+	$repair_list[] = array(
+		'bp-repair-unread-count',
```
Comment (mehulkaklotar) [inline src/bp-messages/bp-messages-filters.php:668]:
```suggestion
		'bp-repair-messages-unread-count',
```

**Comment by mohit-buddyboss** [comment]
@chetansatasiya 
tested this by sending messages and deleting messages in the conversation. This is a client-specific issue and the issue is not reproducible on the dev branch. Although the setting under the tool to fix the count of messages is working fine.
Video: https://www.loom.com/share/842f9291ae9c465880817475379b57aa

@mehulkaklotar

**Pull request merged**